uwh-overlay-drawing/src/main.rs

```diff
@@ -0,0 +1,181 @@
+use std::{str::FromStr, sync::mpsc::channel};
+//use uwh_common::game_snapshot::GamePeriod;
+use network::State;
+use std::net::IpAddr;
+
+use macroquad::prelude::*;
+use uwh_common::game_snapshot::GamePeriod;
+mod load_images;
+mod network;
+mod pages_alpha;
+mod pages_color;
+
+fn window_conf() -> Conf {
+    Conf {
+        window_title: "UWH Overlay".to_owned(),
+        window_width: 1920,
+        window_height: 1080,
+        window_resizable: false,
+        ..Default::default()
+    }
+}
+
+pub struct AppConfig {
+    refbox_ip: IpAddr,
+    refbox_port: u64,
+    uwhscores_url: String,
+}
+
+impl AppConfig {
+    /// Get the app config's refbox ip.
+    pub fn refbox_ip(&self) -> IpAddr {
+        self.refbox_ip
+    }
+
+    /// Get the app config's refbox port.
+    pub fn refbox_port(&self) -> u64 {
+        self.refbox_port
+    }
+
+    /// Get a reference to the app config's uwhscores url.
+    pub fn uwhscores_url(&self) -> &str {
+        self.uwhscores_url.as_ref()
+    }
+}
+
+impl Default for AppConfig {
+    fn default() -> Self {
+        AppConfig {
+            refbox_ip: IpAddr::from_str("127.0.0.1").unwrap(),
+            refbox_port: 8000,
+            uwhscores_url: String::from("uwhscores.com"),
+        }
+    }
+}
+
+#[macroquad::main(window_conf)]
+async fn main() {
+    let (tx, rx) = channel::<State>();
+    let mut conf_directory = home::home_dir().unwrap();
+    conf_directory.push(".config/uwh-overlay/config.json");
+    let config = if let Ok(file) = std::fs::read(conf_directory) {
+        let data: serde_json::Value = serde_json::from_str(std::str::from_utf8(&file).unwrap())
+            .expect("Ill formatted config file!");
+        AppConfig {
+            refbox_ip: IpAddr::from_str(data["refbox_ip"].as_str().unwrap()).unwrap(),
+            refbox_port: data["refbox_port"].as_u64().unwrap(),
+            uwhscores_url: data["uwhscores_url"].as_str().unwrap().to_string(),
+        }
+    } else {
+        AppConfig::default()
+    };
+    std::thread::spawn(|| {
+        network::networking_thread(tx, config)
+            .expect("Networking error. Does the supplied URL exist and is it live?")
+    });
+    let args: Vec<String> = std::env::args().collect();
+    let mut animation_counter = 0f32;
+    let mut secondary_animation_counter = 0f32; // for when we call two draw functions with animations, we need two counters
+    assert!(
+        args.len() == 2,
+        "Got {} args instead of one. Pass one argument, --color or --alpha to get the color or alpha feed respectively",
+        args.len() - 1
+    );
+    let (textures, is_alpha_mode) = if args[1] == *"--color" {
```

Comment:
Yes, the plan is to make the app create both video streams on startup. It's like this right now because having it like this lets me bother with only the stream I'm working on (alpha or colour), and because setting up both screens under a single app would be non trivial.

Non trivial because `Macroquad` doesn't support rendering outside of the main thread : /. So you can't create two windows directly. It would have been simple with the earlier implementation in WGPU, but text rendering proved incredibly difficult to implement manually. I'll have to move networking to the main thread and have it spawn two processes for rendering the streams and implement IPC using sockets or something. This way there will be only one connection to the refbox. Won't be that hard, but I think it would be better done after I'm done with all the pages.

Reply:
I would have to test this, but it's possible that both displays could be drawn to by a single window of sufficient size (ie 3840x1080). Even if it works with a window, it might not work when fullscreen

Reply:
Maybe we don't need this anymore due to [0b54203](https://github.com/AtlantisSports/uwh-refbox-rs/pull/94/commits/0b54203c7bc6bfcaa1c8f7ae281312fa7cbbf4a2), but I just tested, and it's pretty easy to get `i3wm` to stretch a fullscreen window across both screens. Seems like the single window method would be a simple and reliable way to solve this problem, but the dev experience might not be as good on a smaller display.